AdyenActions/Components/Voucher/VoucherComponent.swift

```diff
@@ -49,68 +55,143 @@ public final class VoucherComponent: AnyVoucherActionHandler, Cancellable {
+        voucherViewControllerProvider: AnyVoucherShareableViewProvider?,
+        style: VoucherComponentStyle,
+        passProvider: AnyAppleWalletPassProvider?
+    ) {
+        self.apiContext = apiContext
+        self.style = style
+        self.viewControllerProvider = voucherViewControllerProvider ??
+            VoucherShareableViewProvider(style: style, environment: apiContext.environment)
+        self.passProvider = passProvider ?? AppleWalletPassProvider(apiContext: apiContext)
     }
 
     /// :nodoc:
-    public func didCancel() {
-        passProvider = nil
-    }
+    public func didCancel() { }
 
     /// Handles await action.
     ///
     /// - Parameter action: The await action object.
     public func handle(_ action: VoucherAction) {
+        self.action = action
+        
         Analytics.sendEvent(component: componentName, flavor: _isDropIn ? .dropin : .components, context: apiContext)
-        fetchAndCacheAppleWalletPassIfNeeded(with: action)
+        fetchAndCacheAppleWalletPassIfNeeded(with: action.anyAction)
 
-        var viewControllerProvider = voucherViewControllerProvider
-            ?? VoucherViewControllerProvider(style: style, environment: apiContext.environment)
         viewControllerProvider.localizationParameters = localizationParameters
-        viewControllerProvider.delegate = self
 
-        let viewController = viewControllerProvider.provide(with: action)
+        let view = VoucherView(model: viewModel(with: action))
+        view.delegate = self
+        let viewController = ADYViewController(view: view)
+        view.presenter = viewController
 
         if let presentationDelegate = presentationDelegate {
-            let presentableComponent = PresentableComponentWrapper(component: self, viewController: viewController)
+            let presentableComponent = PresentableComponentWrapper(
+                component: self,
+                viewController: viewController,
+                navBarType: getNavBarType()
+            )
             presentationDelegate.present(component: presentableComponent)
         } else {
             let message = "PresentationDelegate is nil. Provide a presentation delegate to VoucherComponent."
             AdyenAssertion.assertionFailure(message: message)
         }
     }
-
-    private func fetchAndCacheAppleWalletPassIfNeeded(with action: VoucherAction) {
-        if let passToken = action.passCreationToken {
-            passProvider = createPassProvider()
-
-            passProvider?.provide(with: passToken, completion: { _ in /* Do nothting, this is just to cache the response */ })
-        } else {
-            passProvider = nil
+    
+    private func getNavBarType() -> NavigationBarType {
+        let bundle = Bundle(for: UIBarButtonItem.self)
+        let localizedEdit = bundle.localizedString(forKey: "Edit", value: "Edit", table: nil)
+        let localizedDone = bundle.localizedString(forKey: "Done", value: "Done", table: nil)
```

Comment:
This supposed to be translated, right?

Reply:
well, it is translated already :)
i thought it would be nice to use the apple provided translation instead of doing it through Smartling

Reply:
👍